### DISCUSSION TITLE
Vscode jupyter notebooks do not find kernels installed by Pixi

### DISCUSSION THREAD

**Discussion by pakiessling** [prefix-dev/pixi#2642] [Q&A] [answered]
Hi,
absolutely love Pixi!
However I have one small issue.

VScode does not seem to find my pixi enviroments in Jupyter notebooks.

This is a fresh pixi enviroment with ipykernel and pandas in it:
![grafik](https://github.com/user-attachments/assets/3c41ecad-31df-4db6-8129-885ced45ba1d)

I can work around this by doing

```
pixi shell
ipython kernel install --user --name=my_pixi
```

Then I can see the Kernel in the category Jupyter Kernels.
However, from what I have seen this should not be necessary?
Am I doing something wrong?

Thank you!

```
OSX 14.4.1
Visual Studio Code - Insiders 1.96
Python pre-release 2024.21.2024112901
Pixi VSCode 1.0.4
```

**Comment by ruben-arts** [comment]
I'm personally not a big user of Jupyter, pinging @jjjermiah and @renan-r-santos as they are the maintainers of Pixi VSCode and [pixi-kernel](https://github.com/renan-r-santos/pixi-kernel) Which might be of help to you!

**Comment by jjjermiah** [comment]
@pakiessling could you share your `pixi.toml` manifest please?

**Reply by pakiessling** [reply]
Sure

```
[project]
authors = ["pakiessling <user@mail.com>"]
channels = ["conda-forge"]
description = "Add a short description here"
name = "metadata"
platforms = ["osx-arm64"]
version = "0.1.0"

[tasks]

[dependencies]
pandas = ">=2.2.3,<3"
ipykernel = ">=6.29.5,<7"
pip = ">=24.3.1,<25"
openpyxl = ">=3.1.5,<4"
```

**Reply by jjjermiah** [reply, marked as answer]
couple considerations:
- after running pixi shell, can you try opening command palette and clearing the python env cache 
![image](https://github.com/user-attachments/assets/32fe8ef4-b71d-4911-af00-bd83e890b6cf)
 
then try again?

**Reply by pakiessling** [reply]
Hm after clearing the cache and once again entering `pixi shell` the enviroment suddenly appeared as "recommended". Interesting...

**Comment by herrgahr** [comment]
Hi there! Leaving this for everyone who still faces this problem and finds this thread via web search...

If your VS Code still can't find kernels, you may be using Code OSS instead of the official binary published by Microsoft.

In this case, you will probably see errors in the "Python" output panel and / or the "Jupyter" output panel, telling you you have to enable some "proposed" api via a command line switch.

The TL;DR of how to do this is to create the file ~/.config/code-flags.conf with the following contents:

```
--enable-proposed-api=ms-python.python
--enable-proposed-api=ms-toolsai.jupyter
```

> Note: the filename `code-flags.conf` is corrent for Code OSS on Arch Linux, this may be different for your OS / distribution

**Comment by GitHunter0** [comment]
Hey folks @jjjermiah , @renan-r-santos , please, which one of these VScode extensions is the official one?
https://github.com/renan-r-santos/pixi-code 
https://github.com/jjjermiah/pixi-vscode
